hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestCOWDataSource.scala

```diff
@@ -643,8 +736,9 @@ class TestCOWDataSource extends HoodieSparkClientTestBase with ScalaAssertionSup
   def testSparkPartitionByWithCustomKeyGenerator(recordType: HoodieRecordType): Unit = {
     val (writeOpts, readOpts) = getWriterReaderOpts(recordType)
 
+    val updatedWriteOpts = writeOpts.-(KeyGeneratorOptions.PARTITIONPATH_FIELD_NAME.key())
```

Comment:
some cases we need to remove PARTITIONPATH_FIELD_NAME and some cases we don't. this test logic will be hard to maintain. can we avoid special handling like this? as CI passed, we can land once the test is fixed and passed locally.

Reply:
in all tests where we are testing df.partitionBy, we don't want to override partition field and hence. bcoz, the main purpose of these test is, hudi should infer df.partitionBy columns if partition path config is not explicitly overriden by the user. getWriterReaderOpts() unfortunately already has partition path set and hence for these tests I had to remove them. I can add a comment here explaining why we are doing this.